recipe/build.sh

```diff
@@ -8,6 +8,6 @@ ${PYTHON} -c 'import pysr; pysr.install()'
 
 ls ${PREFIX}/share/julia
 
-rm -rf ${PREFIX}/share/julia/!(packages|artifacts)
+rm -rf '${PREFIX}/share/julia/!(packages|artifacts)'
```

Comment:
This line does not seem to do what you want.

```
+ rm -rf '${PREFIX}/share/julia/!(packages|artifacts)'
+ ls /home/conda/feedstock_root/build_artifacts/pysr_1661794169440/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/share/julia
registries
scratchspaces
stdlib
test
artifacts
base
base.cache
clones
compiled
conda
environments
julia-config.jl
logs
packages
prefs
registries
scratchspaces
stdlib
test
```

Reply:
```suggestion
mkdir -p "${PREFIX}/share/SymbolicRegression.jl/depot"
mv "${PREFIX}/share/julia/packages" "${PREFIX}/share/SymbolicRegression.jl/depot/packages"
mv "${PREFIX}/share/julia/artifacts" "${PREFIX}/share/SymbolicRegression.jl/depot/artifacts"
rm -rf "${PREFIX}/share/julia"
```

This probably should be moved into a SymbolicRegression.jl-feedstock, and we should probably try to have an independent PyCall.jl-feedstock.

Reply:
Note that Julia depots stack just like Julia environments stack. The mechanism is used for system installed packages, which is kind of what we are doing.

We probably need the `environments` folder: `share/julia/environments/pysr-0.10.1` and/or need to activate that environment.

Basically we need to satisfy
https://github.com/MilesCranmer/PySR/blob/c3dc203265816db4df8d38e7ea29419eafdb2778/pysr/julia_helpers.py#L87

The last expression here should return `true`.
```python
    from julia.core import JuliaInfo, UnsupportedPythonError
    info = JuliaInfo.load(julia="julia")
    info.is_pycall_built()
```

`is_pycall_built` is from here:
https://github.com/JuliaPy/pyjulia/blob/87c669e2729f9743fe2ab39320ec9b91c9300a96/src/julia/juliainfo.py#L134-L135

Reply:
Looks like there is a `share/julia/conda` which is important.

```
$ ls -R share/julia/conda
share/julia/conda:
3  deps.jl

share/julia/conda/3:

$ cat share/julia/conda/deps.jl 
const ROOTENV = "/home/mkitti/mambaforge/envs/pysr/share/julia/conda/3"
const MINICONDA_VERSION = "3"
const USE_MINIFORGE = true
const CONDA_EXE = "/home/mkitti/mambaforge/envs/pysr/share/julia/conda/3/bin/conda"
```

Without that folder, then PyCall.jl cannot compile.
```julia
julia> using PyCall
[ Info: Precompiling PyCall [438e738f-606a-5dbb-bf0a-cddfbfd45ab0]
ERROR: LoadError: ArgumentError: Path to conda environment is not valid: /home/mkitti/mambaforge/envs/pysr/share/julia/conda/3
Stacktrace:
 [1] prefix(path::String)
   @ Conda ~/mambaforge/envs/pysr/share/pysr_depot/packages/Conda/x2UxR/src/Conda.jl:42
 [2] top-level scope
   @ ~/mambaforge/envs/pysr/share/pysr_depot/packages/Conda/x2UxR/src/Conda.jl:47
 [3] include
   @ ./Base.jl:418 [inlined]
 [4] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
   @ Base ./loading.jl:1318
 [5] top-level scope
   @ none:1
 [6] eval
   @ ./boot.jl:373 [inlined]
 [7] eval(x::Expr)
   @ Base.MainInclude ./client.jl:453
 [8] top-level scope
   @ none:1
in expression starting at /home/mkitti/mambaforge/envs/pysr/share/pysr_depot/packages/Conda/x2UxR/src/Conda.jl:1
ERROR: LoadError: Failed to precompile Conda [8f4d0f93-b110-5947-807f-2305c1781a2d] to /home/mkitti/mambaforge/envs/pysr/share/julia/compiled/v1.7/Conda/jl_gZtz9C.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
    @ Base ./loading.jl:1466
  [3] compilecache(pkg::Base.PkgId, path::String)
    @ Base ./loading.jl:1410
  [4] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1120
  [5] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:1013
  [6] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:997
  [7] include
    @ ./Base.jl:418 [inlined]
  [8] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1318
  [9] top-level scope
    @ none:1
 [10] eval
    @ ./boot.jl:373 [inlined]
 [11] eval(x::Expr)
    @ Base.MainInclude ./client.jl:453
 [12] top-level scope
    @ none:1
in expression starting at /home/mkitti/mambaforge/envs/pysr/share/pysr_depot/packages/PyCall/ygXW2/src/PyCall.jl:1
ERROR: Failed to precompile PyCall [438e738f-606a-5dbb-bf0a-cddfbfd45ab0] to /home/mkitti/mambaforge/envs/pysr/share/julia/compiled/v1.7/PyCall/jl_QiBAvQ.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
   @ Base ./loading.jl:1466
 [3] compilecache(pkg::Base.PkgId, path::String)
   @ Base ./loading.jl:1410
 [4] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1120
 [5] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:1013
 [6] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:997
```

Reply:
That deps.jl is without our recent activate.sh modifications. With the modifications, we should end up with the following, which looks better.

```julia
const ROOTENV = "/home/mkitti/mambaforge/envs/pysr"
const MINICONDA_VERSION = "3"
const USE_MINIFORGE = true
const CONDA_EXE = "/home/mkitti/mambaforge/bin/conda"
```